src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -150,12 +148,15 @@ public void LogEvent(string eventName, string functionName = null, string data =
                         Data = data
                     };
                 },
-                (name, evtToUpdate) =>
+                (name, existing) =>
                 {
-                    // update the existing event
-                    evtToUpdate.Count++;
-
-                    return evtToUpdate;
+                    return new SystemMetricEvent
```

Comment:
When perf testing your changes I was getting failures. The load test failed due to failed requests. I just added the full-ci label to this PR to force a test run, and I expect some tests to fail. One issue is this code - it needs to copy just the FunctionName,EventName,Data,Count properties as in the create statement above. This isn't an aggregate event, it's a "count" event based on event name, so needs these properties set. There are other test failures as well - recommend you run the Metrics unit tests locally to find them all.

Reply:
Even after experimenting myself and fixing things up so tests pass on my box https://github.com/Azure/azure-functions-host/tree/metrics-improvements under load these changes aren't working (request failures).

Reply:
Oh I see, sorry for wasting your cycles @mathewc Will run it locally and push fixes soon.